src/API/IPFS.ts

```diff
@@ -0,0 +1,22 @@
+import { inject, injectable } from 'inversify'
+import fetch from 'node-fetch'
+
+export interface IPFSConfiguration {
+  readonly ipfsUrl: string
+}
+
+type checkHealth = () => Promise<number>
+
+@injectable()
+export class IPFS {
+  private readonly url: string
+
+  constructor(@inject('IPFSConfiguration') configuration: IPFSConfiguration) {
+    this.url = configuration.ipfsUrl
+  }
+  
+  checkHealth: checkHealth = async () => {
```

Comment:
It's generally better for application clients to match the API one-to-one, and have our business logic in another layer (controller usually).

IOW, exposing a `getVersion` here, and doing the ipfsIsConnected logic in the controller.

Since you're relying on the status, which shouldn't leave the IPFS client, maybe `throw` if `status !== 200`. I believe `fetch` already throws for network errors, so if mongo is unavailable even to respond with a `500`, we'll get an error thrown anyway.

Reply:
Ok I changed this up. I'm not sure I understood what you mean in the last paragraph though.